packages/cli/README.md

```diff
@@ -127,3 +127,615 @@ jobs:
+why runner function can't be used...
+
+We will implement a performance focused plugin using the [Lighthouse CLI](https://github.com/GoogleChrome/lighthouse#using-the-node-cli) as real life example.
+
+Let's start with a `crate` function maintaining the basic information of the `PluginConfig`.
+
+<details>
+<summary> <b>lighthouse plugin setup</b> (collapsed for brevity) </summary>
+
+```typescript
+// code-pushup.config.ts
+import { PluginConfig, RunnerConfig } from '@code-pushup/models';
+
+type Options = {
+  url: string;
+  onlyAudits: string;
+};
+
+async function create(options: Options): Promise<PluginConfig> {
+  const lcpAudit = {
+    slug: 'largest-contentful-paint',
+    title: 'Largest Contnentful Paint',
+  };
+
+  return {
+    slug: 'lighthouse',
+    title: 'Chrome Lighthouse plugin',
+    icon: 'lighthouse',
+    audits: [
+      lcpAudit,
+      // other lighthouse audits
+    ],
+    runner: runnerConfig(options),
+  };
+}
+
+function runnerConfig(options: Options): RunnerConfig {
+  // implementation follows in the below section
+  return {} as RunnerConfig;
+}
+
+export default {
+  // ...
+  plugins: [
+    await create({
+      url: 'https://example.com',
+      onlyAudits: 'largest-contentful-paint',
+    }),
+  ],
+  // ...
+};
+```
+
+</details>
+
+The first thing we should think about is how to get the raw data.
+With the lighthouse CLI it is ease as it already provides a report file in `json` format containing a set of audits.
+
+The lighthouse CLI can be executed like this: `npx lighthouse https://example.com`  
+You should see console output of the audits created by the CLI.
+
+To get better debugging experience we add a couple of more options:
+
+- The format and location of the output can be configured with `--output=json --outputFile=lighthouse-report.json`.  
+  This ensures we avoid overwrites of other existing files and is needed to be able to load the generated lighthouse
+  report.
+- To reduce the output you can execute only specific audits with the `--onlyAudits` option
+  e.g.: `--onlyAudits=largest-contentful-paint`.  
+  This will significantly reduce the time lighthouse takes to run.
+- If we want to run the script in the background we can execute lighthouse headless with the
+  flag `--chrome-flags="--headless=new"`.  
+  It also is helpful when executing lighthouse in th CI.
+
+The basic implementation of a `RunnerConfig` for the above command looks like this:
+
+```typescript
+// code-pushup.config.ts
+// ...
+import { join } from 'path';
+import { AuditOutputs } from '@code-pushup/models';
+
+function runnerConfig(options: Options): RunnerConfig {
+  const { url, onlyAudits } = options;
+  const outputFile = join(process.cwd(), '.code-pushup', 'lighthouse-report.json');
+  return {
+    // npx lighthouse https://example.com --output=json --outputFile=lighthouse-report.json  --onlyAudits=largest-contentful-paint
+    command: 'npx',
+    args: objectToCliArgs({
```

Comment:
You should include the import from `@code-pushup/utils`, since this helper hasn't been established before.